pkg/proxy/iptables/proxier_test.go

```diff
@@ -2481,128 +2481,253 @@ func TestLoadBalancer(t *testing.T) {
+			localhostNodePorts: false,
+			nodePortAddresses:  []string{"192.168.0.0/24", "127.0.0.1/32"},
+
+			allowAltNodeIP: false,
+			expectFirewall: false,
 		},
 		{
-			name:     "external to nodePort on secondary IP",
-			sourceIP: testExternalClient,
-			destIP:   testNodeIPAlt,
-			destPort: svcNodePort,
-			output:   fmt.Sprintf("%s:%d", epIP, svcPort),
-			masq:     true,
+			name: "ipv4, localhost-nodeports enabled, multiple nodeport-addresses",
+
+			family:             v1.IPv4Protocol,
+			localhostNodePorts: false,
+			nodePortAddresses:  []string{"192.168.0.0/24", "192.168.1.0/24", "2001:db8::/64"},
+
+			allowAltNodeIP: true,
+			expectFirewall: false,
 		},
 		{
-			name:     "node to nodePort",
-			sourceIP: testNodeIP,
-			destIP:   testNodeIP,
-			destPort: svcNodePort,
-			output:   fmt.Sprintf("%s:%d", epIP, svcPort),
-			masq:     true,
+			name: "ipv6, localhost-nodeports enabled",
+
+			family:             v1.IPv6Protocol,
+			localhostNodePorts: true,
+			nodePortAddresses:  nil,
+
+			allowAltNodeIP: true,
+			expectFirewall: false,
 		},
 		{
-			name:     "localhost to nodePort gets masqueraded",
-			sourceIP: "127.0.0.1",
-			destIP:   "127.0.0.1",
-			destPort: svcNodePort,
-			output:   fmt.Sprintf("%s:%d", epIP, svcPort),
-			masq:     true,
+			name: "ipv6, localhost-nodeports disabled",
+
+			family:             v1.IPv6Protocol,
+			localhostNodePorts: false,
+			nodePortAddresses:  nil,
+
+			allowAltNodeIP: true,
+			expectFirewall: false,
 		},
-	})
+		{
+			name: "ipv6, localhost-nodeports disabled, multiple nodeport-addresses",
+
+			family:             v1.IPv6Protocol,
+			localhostNodePorts: false,
+			nodePortAddresses:  []string{"192.168.0.0/24", "192.168.1.0/24", "2001:db8::/64"},
+
+			allowAltNodeIP: false,
+			expectFirewall: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			var ipt *iptablestest.FakeIPTables
+			var svcIP, epIP1, epIP2 string
+			if tc.family == v1.IPv4Protocol {
+				ipt = iptablestest.NewFake()
+				svcIP = "172.30.0.41"
+				epIP1 = "10.180.0.1"
+				epIP2 = "10.180.2.1"
+			} else {
+				ipt = iptablestest.NewIPv6Fake()
+				svcIP = "fd00:172:30::41"
+				epIP1 = "fd00:10:180::1"
+				epIP2 = "fd00:10:180::2:1"
+			}
+			fp := NewFakeProxier(ipt)
+			fp.localhostNodePorts = tc.localhostNodePorts
+			if tc.nodePortAddresses != nil {
+				fp.nodePortAddresses = proxyutil.NewNodePortAddresses(tc.family, tc.nodePortAddresses)
+			}
+
+			makeServiceMap(fp,
+				makeTestService("ns1", "svc1", func(svc *v1.Service) {
+					svc.Spec.Type = v1.ServiceTypeNodePort
+					svc.Spec.ClusterIP = svcIP
+					svc.Spec.Ports = []v1.ServicePort{{
+						Name:     "p80",
+						Port:     80,
+						Protocol: v1.ProtocolTCP,
+						NodePort: 3001,
+					}}
+				}),
+			)
+
+			populateEndpointSlices(fp,
+				makeTestEndpointSlice("ns1", "svc1", 1, func(eps *discovery.EndpointSlice) {
+					eps.AddressType = discovery.AddressTypeIPv4
```

Comment:
does not depend on the `tc.Family`?

Reply:
is this test passing with the wong AddressType?

Reply:
Huh... yeah, in the context of these unit tests, AddressType never gets checked; all that matters is the actual IP family of the endpoint addresses. (If there was a MetaProxier involved, then AddressType would have to be correct for the EndpointSlice to get routed to the correct single-stack `iptables.Proxier`, but since we only have one proxier here and manually feed EndpointSlices to it, AddressType doesn't matter.)

Reply:
can we change it to add the right family just for correctness and to avoid someone in the future freak out seeing this?